src/lib/OpenEXR/ImfHeader.cpp

```diff
@@ -60,8 +62,56 @@ using IMATH_NAMESPACE::Box2i;
 using IMATH_NAMESPACE::V2i;
 using IMATH_NAMESPACE::V2f;
 
+namespace
+{
+
+static int   s_DefaultZipCompressionLevel = Z_DEFAULT_COMPRESSION;
+static float s_DefaultDwaCompressionLevel = 45.f;
+static std::mutex s_globCompressionMutex;
+struct CompressionRecord
+{
+    CompressionRecord()
+        : zip_level (s_DefaultZipCompressionLevel),
+          dwa_level (s_DefaultDwaCompressionLevel)
+    {}
+    int zip_level;
+    float dwa_level;
+};
+static std::map<const void *, CompressionRecord> s_globCompressionStore;
```

Comment:
I never really thought about this, but oh my, the pains/workarounds one has to go through to avoid breaking C++ ABI are sad. One more reason to avoid C++ based public interfaces, I guess. If everything was behind opaque types with a C-like interface (very much like the new OpenEXRCore), none of this would have to be done.

Reply:
Well, to be fair, it's how we did it, not necessarily C vs C++ per se. The original C++ could have used a PIMPL idiom to prevent exposure of the internal data layouts, and a C API can still unwisely expose structure internals.